v1/oneplus2.json

```diff
@@ -0,0 +1,294 @@
+{
+  "name": "Oneplus 2",
+  "codename": "oneplus2",
+  "unlock": ["confirm_oemunlock", "update", "twrp"],
+  "user_actions": {
+    "confirm_oemunlock": {
+      "title": "Unlock your bootloader",
+      "description": "Please make sure the bootloader is unlocked",
+      "link": "https://androidmtk.com/unlock-bootloader-oneplus-2"
```

Comment:
That just uses `fastboot oem unlock`, the installer should be able to handle that already.

Reply:
I haven't found example for the unlocking part
is it something like that?
```
"steps": [
      {
          "type": "fastboot:oem_unlock"
      }
]
```

Reply:
I see there is some kind of lock detection (src/main.js line 342) but I don't know how to trigger/manage it :/

Reply:
Lock your device (if it supports it `sudo oem lock`) and run the installer again. The `fastboot:flash` step will then run into an error that [will be recognized](https://github.com/ubports/promise-android-tools/blob/master/src/fastboot.js#L48) by our library. That'll cause the installer to jump out of the install step chain to prompt the user for permission to unlock. If the user clicks ok, the installer will run `fastboot oem unlock` and resume the install at the last fastboot flash step. Usually very reliable. There are many different errors devices can throw (because there's no standardization in fastboot), but if i get a report for a new unhandled error one, i know how to proceed and can fix it in the next installer release.

As a way forward, i propose just removing the instruction. It should usually just work.

Reply:
Take care, locking and unlocking the bootloader might wipe your data ;)

Reply:
@NeoTheThird there is an issue with "lock detection"
the error returned by the bootloader is different from the one in [fastboot.js](https://github.com/ubports/promise-android-tools/blob/master/src/fastboot.js#L48)
```
Error: fastboot:flash: Error: Flashing failed: {"error":{"code":1},"stderr":"FAILED (remote: 'device is locked. Cannot flash images')fastboot: error: Command failed"}
```
NOTE: Settings > Developer Options > OEM unlocking is enabled

@Flohack74 Yes, unlocking the bootloader delete /data... I'll add a warning about that in thee oneplus2 config

Reply:
@Vince1171 no, don't add a warning, that's default behavior. I'll add handling for that error. If you want, you can leave the oem unlock note in for now, but we can remove it again soon.

Reply:
@NeoTheThird  I've changed the message
let's wait for the merge of ubports/promise-android-tools#52 before merging this one

Reply:
>  I've changed the message

ok. i'd have removed it completely, as there's generic instructions for this, but it's ok for me if you want to keep it like this.

>  let's wait for the merge of ubports/promise-android-tools#52 before merging this one

As you wish. I'll try to squeeze in another 0.7 installer release with that included before entering QA for 0.8, so you don't have to wait a week.

Reply:
There you go, a prerelease: https://github.com/ubports/ubports-installer/releases/tag/0.7.4-beta

Reply:
done already, I was quicker :P
It works perfectly thank you :D

Reply:
and I've remove the oem unlock message